fix(netlify): only cache GET/HEAD methods (#127)
withastro/adapters: .changeset/dry-rules-bathe.md

---
'@astrojs/netlify': patch
---

> only cache GET/HEAD requests with cacheOnDemandPages

Reviewer: Can we try to reword this changeset to follow our sematics, e.g. `Fixes ... to ..., Fixes ... which ...`

Some guidance questions, not mandatory to answer all, but helpful.
What does this change do?
How does this change do it?
Why is this change necessary?

Author: Made an update in https://github.com/withastro/adapters/pull/127/commits/277fc09d372392109850c29e8c4691df93a1602c, how do you like it?

> Fixes cacheOnDemandPages setting to only cache GET/HEAD requests by default

Reviewer 2: Just a tiny suggestion, let me know what you guys think 😁
```suggestion
Enhanced caching behavior for the `cacheOnDemandPages` setting. Now, it explicitly caches only GET and HEAD requests by default.
```

Reviewer: Thanks for the suggestion, I think it is good to add some more details, like you did. However we as maintainers unofficially agreed to use the present tense, so I suggest:

Enhances caching behavior for the `cacheOnDemandPages` setting. Now, it explicitly caches only GET and HEAD requests by default.

> Fixes cacheOnDemandPages setting to only cache GET/HEAD requests by default

Reviewer 3: ```suggestion
Updates the behavior of the `cacheOnDemandPages` setting to only cache GET/HEAD requests by default
```
Maybe a slight preference for "updates the behaviour" as presenting it more of a "thing you might want to check if you have this enabled because it changed" vs just saying "fix" (which more implies something that was broken before should now work, not "we changed how it worked").

Just a thought! Up to you! Either way, I would use code syntax for the property!

> Fixes cacheOnDemandPages setting to only cache GET/HEAD requests by default

Reviewer 4: Since I'm here, I'll throw my hat into the ring! I agree with "updates", as nothing broke before!

```suggestion
Updates the default behavior of `cacheOnDemandPages` to cache only GET/HEAD requests 
```
Or you can place "default" back at the end. Up to you!
